src/torch/README.md

```diff
@@ -0,0 +1,123 @@
+# SIRF-PyTorch Wrapper
+This wrapper provides a bridge between the [SIRF](https://github.com/SyneRBI/SIRF) (Synergistic Image Reconstruction Framework) library and PyTorch, enabling the use of SIRF's image reconstruction operators and objective functions within PyTorch's automatic differentiation (autodiff) framework. 
+
+## Core Concepts: Autodiff, Reverse Mode, JVP, VJP, and HVP
+
+This wrapper leverages PyTorch's `torch.autograd.Function` to achieve integration with SIRF.  Understanding the following concepts is crucial for the design:
+
+*   **Automatic Differentiation (Autodiff):** A technique for numerically evaluating the derivative of a function specified by a computer program.  PyTorch uses a "reverse mode" autodiff system.
+
+*   **Reverse Mode Autodiff:**  In reverse mode, the gradients are computed by traversing the computational graph backwards, from the output (typically a scalar loss function) to the inputs.  This is highly efficient for functions with many inputs and a single output, which is common in machine learning.
+
+*   **Jacobian-Vector Product (JVP):**  Let `y = f(x)` be a vector-valued function, where `x` is an input vector and `y` is an output vector. The Jacobian matrix `J` contains all the first-order partial derivatives of `f`.  The JVP is the product `Jv`, where `v` is a vector.  It gives the directional derivative of `f` at `x` in the direction of `v`.  *Forward mode* autodiff computes JVPs.
```

Comment:
"It gives the directional derivative of `f` at `x` in the direction of `v`. " I think that strictly speaking this is only true for unit vectors. @mehrhardt could confirm.

(same comment for other statements below)

Reply:
Yes I think you're right too now... Perhaps "scaled" direction derivative, or change in f in direction of v?